packages/flutter/test/widgets/framework_test.dart

```diff
@@ -1574,6 +1574,31 @@ void main() {
     expect(RawKeyboard.instance.keyEventHandler, same(rawKeyEventHandler));
   });
 
+  testWidgets('Incorrect use of ParentDataWidget shows a useful error message', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      SizedBox(
+        key: const Key('container'),
+        child: Flexible( // Flexible should not be used directly in a Container or other inappropriate parents
+          child: Container(),
```

Comment:
This is still a `Container`. This should probably also be a `SizedBox()`. See https://dart.dev/tools/linter-rules/avoid_unnecessary_containers

You might also need to add `const` to the first SizedBox widget.